scripts/gatsby-plugin-checker/index.js

```diff
@@ -64,6 +64,60 @@ const filterNotNotified = (packages, plugins) => {
   return packages.filter(pkg => notified.indexOf(pkg.name) < 0)
 }
 
+const removePackagesWithoutRepository = packages =>
+  packages.filter(p => hasRepository(p))
+
+const hasRepository = packageToCheck => {
+  if (packageToCheck.links.repository) {
+    return true
+  }
+  return false
+}
+
+const removePackagesWithBadRepoLinks = async packages => {
+  let packagesWithValidRepoLinks = []
+  for (let i = 0; i < packages.length; i++) {
+    const hasValidRepo = await hasValidRepository(packages[i])
+    if (hasValidRepo) packagesWithValidRepoLinks.push(packages[i])
+  }
+  return packagesWithValidRepoLinks
+}
+
+const hasValidRepository = async packageToSearch => {
+  const response = got(packageToSearch.links.repository)
+  if (response.statusCode == 404) return false
+  return true
+}
+
+const removeBadNameFormats = packages => packages.filter(p => hasGoodName(p))
+
+const hasGoodName = pkg => {
+  const name = pkg.name
+
+  if (name.indexOf(`/`) !== -1) {
+    const nameWithoutScope = name.split(`/`, 2)[1]
+    if (startsWithAllowedPrefix(nameWithoutScope)) return true
+  } else {
+    if (startsWithAllowedPrefix(name)) return true
+  }
+  return false
+}
+
+const startsWithAllowedPrefix = name => {
+  let isGoodName = false
+  keywords.forEach(keyword => {
+    if (name.indexOf(keyword) === 0) isGoodName = true
+  })
+  return isGoodName
+}
+
+const removePackagesWithoutReadme = packages =>
+  packages.filter(p => hasReadMe(p))
+
+const hasReadMe = pkg => {
+  if (pkg.links.homepage || pkg.readme) return true
```

Comment:
we should try to do some extra work here to check if README.md is found in the repository if it has one like yarn does: https://github.com/yarnpkg/website/blob/master/js/src/lib/Details/index.js#L122-L131 we can just check for a 404 like you do with the git repository

Reply:
I just basically did a check for a 404 by checking if it was in the root folder of the project, not sure if there's a better way to do it, but I figure pretty much everyone puts the README in the root directory.